howtos/diffs/polyak-averaging.diff

```diff
@@ -61,6 +61,21 @@ index 51d2fde..eab7983 100644
 +    polyak_loss, polyak_accuracy = eval_model(model_ema, test_ds)
 +    logging.info('polyak eval epoch: %d, loss: %.4f, accuracy: %.2f',
 +                 epoch, polyak_loss, polyak_accuracy * 100)
-   return optimizer
- 
- 
+     summary_writer.scalar('train_loss', train_metrics['loss'], epoch)
+     summary_writer.scalar('train_accuracy', train_metrics['accuracy'], epoch)
+     summary_writer.scalar('eval_loss', loss, epoch)
+diff --git a/examples/mnist/train_test.py b/examples/mnist/train_test.py
+index fb02770..4cfec2a 100644
+--- a/examples/mnist/train_test.py
++++ b/examples/mnist/train_test.py
+@@ -34,8 +34,8 @@ class TrainTest(absltest.TestCase):
+     input_rng = onp.random.RandomState(0)
+     model = train.create_model(random.PRNGKey(0))
+     optimizer = train.create_optimizer(model, 0.1, 0.9)
+-    optimizer, train_metrics = train.train_epoch(optimizer, train_ds, 128, 0,
+-                                                 input_rng)
++    optimizer, train_metrics = train.train_epoch(optimizer, model.params, train_ds,
```

Comment:
This seems wrong? `train_epoch` does not take a params 2nd argument?

Reply:
(Do tests pass on this HOWTO? I'm surprised)

Reply:
I think this diff of diff might be confusing to read; the signature of train_epoch that this diff of diff references is with respect to the original example, but if you look at the original diff file itself (ie polyak-averaging.diff), the signature of train_epoch changes.

(Ah shoot, on mobile and copied the line not the link to the line):

+def train_epoch(optimizer, params_ema, train_ds, batch_size, epoch, rng):

Reply:
Link: https://github.com/google/flax/blob/dea96db3fa779b25c32ab8c40f4b55a7d26fec93/howtos/diffs/polyak-averaging.diff#L32

Reply:
Ah, I see. This is why the tests weren't passing originally.